ios/FluentUI/Shimmer/ShimmerLinesView.swift

```diff
@@ -13,81 +13,120 @@ public typealias MSShimmerLinesView = ShimmerLinesView
-
-            currentTop += shimmerLinesViewAppearance.lineHeight + shimmerLinesViewAppearance.lineSpacing
-        }
-
-        shimmeringLayer.frame = CGRect(x: -shimmerAppearance.width, y: 0.0, width: frame.width + 2 * shimmerAppearance.width, height: frame.height)
-
-        viewCoverLayers.forEach { $0.frame = flipRectForRTL($0.frame) }
-
-        updateShimmeringLayer()
-        updateShimmeringAnimation()
-    }
-
-    open override func sizeThatFits(_ size: CGSize) -> CGSize {
-        return ShimmerLinesView.sizeThatFits(size, appearance: shimmerLinesViewAppearance)
-    }
-
-    open override var intrinsicContentSize: CGSize {
-        return CGSize(width: UIView.noIntrinsicMetric, height: sizeThatFits(CGSize(width: frame.width, height: .infinity)).height)
-    }
-
-    override func updateViewCoverLayers() {
-        var newLineLayers = [CALayer]()
-        let desiredLineCount = ShimmerLinesView.lineCount(for: shimmerLinesViewAppearance, availableHeight: frame.height)
-
-        for i in 0..<desiredLineCount {
-            let lineLayer = i < viewCoverLayers.count ? viewCoverLayers[i] : CALayer()
-
-            lineLayer.cornerRadius = appearance.labelCornerRadius >= 0 ? appearance.labelCornerRadius : appearance.cornerRadius
-            lineLayer.backgroundColor = appearance.tintColor.cgColor
-
-            // Add layer
-            newLineLayers.append(lineLayer)
-            layer.addSublayer(lineLayer)
-        }
-
-        Set(viewCoverLayers).subtracting(Set(newLineLayers)).forEach { $0.removeFromSuperlayer() }
-
-        viewCoverLayers = newLineLayers
-    }
+
+	@available(*, deprecated, message: "Use individual properties instead")
+	@objc public var shimmerLinesViewAppearance = ShimmerLinesViewAppearance() {
+		didSet {
+			lineCount = shimmerLinesViewAppearance.lineCount
+			lineHeight = shimmerLinesViewAppearance.lineHeight
+			lineSpacing = shimmerLinesViewAppearance.lineSpacing
+			firstLineFillPercent = shimmerLinesViewAppearance.firstLineFillPercent
+			lastLineFillPercent = shimmerLinesViewAppearance.lastLineFillPercent
+
+			setNeedsLayout()
+		}
+	}
+
+	/// Number of lines that will shimmer in this view. Use 0 if the number of lines should fill the available space.
+	@objc open var lineCount: Int = 3 {
+		didSet {
+			setNeedsLayout()
+		}
+	}
+
+	/// Height of shimmering line
+	@objc open var lineHeight: CGFloat = 11 {
+		didSet {
+			setNeedsLayout()
+		}
+	}
+
+	/// Spacing between lines (if lines > 1)
+	@objc open var lineSpacing: CGFloat = 11 {
+		didSet {
+			setNeedsLayout()
+		}
+	}
+
+	/// The percent the first line (if 2+ lines) should fill the available horizontal space
+	@objc open var firstLineFillPercent: CGFloat = 0.94 {
+		didSet {
+			setNeedsLayout()
+		}
+	}
+
+	/// The percent the last line should fill the available horizontal space.
+	@objc open var lastLineFillPercent: CGFloat = 0.6 {
```

Comment:
>0 [](start = 47, length = 1)

should these const be defined separately?

Reply:
Probably, good catch